### ISSUE TITLE
Writing MIDI header manually conflict with Reading header length

### ISSUE THREAD

**Issue by ReDuzed** [naudio/NAudio#867]
Following the file format structure in the heading chunk, I can get NAudio to read the Chunk type -- easy -- but then reading the following Int32 I get ``Unexpected header chunk length``. Reading the file back manually with BinaryReader, br.ReadUInt32(), returns the requisite value of 6.

Here is a direct code sample of the Header chunk writing sequence:
```c#
bw.Write(Encoding.ASCII.GetBytes(HeaderChunk.Type));
bw.Write(HeaderChunk.Length);
bw.Write(HeaderChunk.Format);
bw.Write(HeaderChunk.Track);
bw.Write(HeaderChunk.Division);
```

Previous to posting an issue I noticed SwapUInt32 was being used. I'd rather not swipe that code and walk away.

**Comment by markheath** [comment]
Yes, this is an endianness issue. The SMF spec stores the `MThd` length as **big-endian** (`00 00 00 06`), which is why NAudio uses `SwapUInt32` when reading/writing. `BinaryWriter.Write(6)` emits little-endian bytes (`06 00 00 00`), so it round-trips fine through your own `BinaryReader` but isn't a valid MIDI file — NAudio byte-swaps it to `0x06000000` and rejects it.

Write the length big-endian (e.g. `bw.Write(SwapUInt32(HeaderChunk.Length))`), or just use `MidiFile.Export(...)` instead of writing the chunks by hand.
